### PR TITLE
[SCT-586]Add type

### PR DESCRIPTION
**Pull request by tiramisu24**
add object type to return

### CODE REVIEW

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 1]
```diff
@@ -420,9 +422,113 @@ private SearchObjectsOutput searchObjects(
...
+                .put("type", "Blah")
+                .put("creator", "user")
+                .put("copied", "user2")
+                .put("module", "module")
+                .put("method", "method")
+                .put("module_ver", "2")
+                .put("commit", "commitHash")
+                .build();
+
+        Map<String, String > additonalProperties = actual.getObjectProps();
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: just compare the expected map to the additionalProperties map directly

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 2]
```diff
@@ -420,9 +422,113 @@ private SearchObjectsOutput searchObjects(
...
                 .withPostProcessing(pp)
-                .withMatchFilter(new MatchFilter().withFullTextInAll("test"))
+                .withMatchFilter(new MatchFilter().withFullTextInAll(query))
                 .withAccessFilter(new AccessFilter());
 
         return  sm.searchObjects(input, "auser");
     }
-}
+
+    @Test
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: Doesn't there need to be a case where the type is null?

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 3]
```diff
@@ -420,9 +422,113 @@ private SearchObjectsOutput searchObjects(
...
+
+        assertThat("did not find objects", res1.getObjects().size() == 1, is(true));
+        kbasesearchengine.ObjectData actual = res1.getObjects().get(0);
+
+        assertThat("incorrect data", actual.getData().toString(),
+                is(new UObject("obj").toString()));
+        assertThat("incorrect guid", actual.getGuid(), is(guid.toString()));
+        Map <String, String> expectedKeyProps = new HashMap<>();
+
+        expectedKeyProps.put("key", "prop");
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: Skip keys was true, so shouldn't this be empty?

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 1 above)
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: I thought this was more informative this be more informative?

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 3 above)
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: ops intent was to turn on everything

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 1 above)
Comment (MrCreosote) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: OK. I usually go for simpler code, but I can live with a test per key.

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 2 above)
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: yes damn

File: lib/src/kbasesearchengine/main/SearchMethods.java [hunk 4]
```diff
@@ -212,16 +214,21 @@ private Pagination fromSearch(kbasesearchengine.search.Pagination pg) {
         if (od.getParentData().isPresent()) {
             ret.withParentData(new UObject(od.getParentData().get()));
         }
+
         ret.withObjectName(od.getObjectName().orNull());
         ret.withKeyProps(od.getKeyProperties());
         ret.withHighlight(od.getHighlight());
 
+        SearchObjectType type = od.getType().orNull();
+        if(type != null){
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/main/SearchMethods.java]: Better practice is 
```
if (od.getType().isPresent()) {
    // do stuff
}

**Pull request merged**